Won't compile with qt 6.5

Error (from VCPKG CI):
```
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(98): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(98): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(101): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(101): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(104): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(104): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(107): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(107): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(110): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(110): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(113): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(113): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(116): error C2440: 'return': cannot convert from 'QMetaMethodReturnArgument' to 'QGenericReturnArgument'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(116): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(201): warning C4996: 'QVariant::canConvert': was declared deprecated
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(211): warning C4996: 'QVariant::canConvert': was declared deprecated
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(221): warning C4996: 'QVariant::canConvert': was declared deprecated
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(259): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(260): warning C4996: 'QVariant::canConvert': was declared deprecated
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(271): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(286): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): error C2665: 'std::vector<QGenericArgument,std::allocator<QGenericArgument>>::push_back': no overloaded function could convert all the argument types
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\vector(939): note: could be 'void std::vector<QGenericArgument,std::allocator<QGenericArgument>>::push_back(_Ty &&)'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: 'void std::vector<QGenericArgument,std::allocator<QGenericArgument>>::push_back(_Ty &&)': cannot convert argument 1 from 'QMetaMethodArgument' to '_Ty &&'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: Reason: cannot convert from 'QMetaMethodArgument' to '_Ty'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\vector(935): note: or       'void std::vector<QGenericArgument,std::allocator<QGenericArgument>>::push_back(const _Ty &)'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: 'void std::vector<QGenericArgument,std::allocator<QGenericArgument>>::push_back(const _Ty &)': cannot convert argument 1 from 'QMetaMethodArgument' to 'const _Ty &'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: Reason: cannot convert from 'QMetaMethodArgument' to 'const _Ty'
        with
        [
            _Ty=QGenericArgument
        ]
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(291): note: while trying to match the argument list '(QMetaMethodArgument)'
D:\buildtrees\spix\src\v0.5-97159dbd3f.clean\lib\src\Scene\Qt\QtItemTools.cpp(288): warning C4996: 'QVariant::convert': was declared deprecated
```

--- a/lib/src/Scene/Qt/QtItemTools.cpp
+++ b/lib/src/Scene/Qt/QtItemTools.cpp
@@ -95,25 +95,25 @@ QGenericReturnArgument GetReturnArgForQMetaType(int type, QMLReturnVariant& retV
         return QGenericReturnArgument();
     case QMetaType::Type::Bool:
         retVar = bool();
-        return Q_RETURN_ARG(bool, std::get<bool>(retVar));
+        return QReturnArgument<bool>("bool", std::get<bool>(retVar));
     case QMetaType::Type::Int:
         retVar = int();
-        return Q_RETURN_ARG(int, std::get<int>(retVar));
+        return QReturnArgument<int>("int", std::get<int>(retVar));
     case QMetaType::Type::Float:
         retVar = float();
-        return Q_RETURN_ARG(float, std::get<float>(retVar));
+        return QReturnArgument<float>("float", std::get<float>(retVar));
     case QMetaType::Type::Double:
         retVar = double();
-        return Q_RETURN_ARG(double, std::get<double>(retVar));
+        return QReturnArgument<double>("double", std::get<double>(retVar));
     case QMetaType::Type::QString:
         retVar = QString();
-        return Q_RETURN_ARG(QString, std::get<QString>(retVar));
+        return QReturnArgument<QString>("QString", std::get<QString>(retVar));
     case QMetaType::Type::QDateTime:
         retVar = QDateTime();
-        return Q_RETURN_ARG(QDateTime, std::get<QDateTime>(retVar));
+        return QReturnArgument<QDateTime>("QDateTime", std::get<QDateTime>(retVar));
     default:
         retVar = QVariant();
-        return Q_RETURN_ARG(QVariant, std::get<QVariant>(retVar));
+        return QReturnArgument<QVariant>("QVariant", std::get<QVariant>(retVar));
     }
 }
 
@@ -288,7 +288,7 @@ std::vector<QGenericArgument> ConvertAndCreateQArgumentsForMethod(
                 varargs[i].convert(targetType);
                 qtArgs.push_back(QGenericArgument(varargs[i].typeName(), varargs[i].data()));
             } else {
-                qtArgs.push_back(Q_ARG(QVariant, varargs[i]));
+                qtArgs.push_back(QArgument<QVariant>("QVariant", varargs[i]));
             }
         } else
             qtArgs.push_back(QGenericArgument());

